### PR TITLE
[WIP] Allow `visit()` customization

### LINKED ISSUE #60: Allow overriding or customization of the visit options
Currently, visit API of fastboot is getting invoked with preset options defined in the middleware [here](https://github.com/embermap/ember-cli-fastboot-testing/blob/master/index.js#L68).

It will be useful if `visit` test helper can take custom visit options and pass it to fastboot's visit api. I think this set should per test basis.

@ryanto

### PR DESCRIPTION
**Pull request by CvX**
WIP, needs tests

Closes #60

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,25 +57,29 @@ module.exports = {
...
           url: parsed.path,
           query: parsed.query,
-          headers: headers
+          headers,
         },
-        response: {}
+        response: {},
       };
 
+      let options = Object.assign(defaultOptions, req.body.options);
```
Comment (ryanto) [inline index.js]: I like this, it lets anyone override request/response if they really wanted to.

Comment (dnalagatla) [reply index.js]: I agree. Thank you for adding.

**Comment by dnalagatla** [comment]
@CvX Thank you for looking into this Feature. Any plans for merging this PR soon. Let me know if you need any help here. 

This Feature and #57 will give the flexibility for us to use this addon for SSR testing @ LinkedIn.

Let me know if you have any inputs for #113. Thank you

**Comment by dnalagatla** [comment]
This PR looks good to me and covers the scenarios I was looking for. Any blockers for merging the changes? Thank you @CvX for working on this PR. 

cc: @ryanto

**Comment by CvX** [comment]
Yeah, still gotta finish the inline docs and add tests.

**Comment by dnalagatla** [comment]
Thank you @ryanto

**Pull request merged**